app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt

```diff
@@ -108,26 +112,45 @@ class ProjectInformationActivity : AppCompatActivity() {
 
     private fun setUpApplyButton(applyButton: Button) {
         val projectId = projectVar.id
-        val userDataDatabase = Utils.getInstance(this).userDataDatabase
+        val utils = Utils.getInstance(this)
+        val userDataDatabase = utils.userDataDatabase
+        val candidatureDatabase = utils.candidatureDatabase
         var alreadyApplied = false
-        setApplyButtonText(applyButton,null)
+        setApplyButtonText(applyButton, null)
+
+        val candidature = Candidature(projectId, "dummy", EMPTY_PROFILE, EMPTY_CV,
+            Candidature.State.Waiting)
         userDataDatabase.getListOfAppliedToProjects({ projectIds ->
             alreadyApplied = projectIds.contains(projectId)
             setApplyButtonText(applyButton, alreadyApplied)
-        },{})
+        }, {})
 
         applyButton.isEnabled = !projectVar.isTaken
 
         applyButton.setOnClickListener {
             userDataDatabase.applyUnapply(
                 !alreadyApplied,
-               projectId,
+                projectId,
                 {
                     showToast(getString(R.string.success), Toast.LENGTH_SHORT)
                     alreadyApplied = !alreadyApplied
                     setApplyButtonText(applyButton, alreadyApplied)
+                    if (!alreadyApplied) {
```

Comment:
Because the alreadyApplied was already inverse, the if should not be inversed.
```suggestion
                    if (alreadyApplied) {
```